src/prefect/cli/cloud.py

```diff
@@ -170,53 +376,48 @@ async def login(
-                        exit_with_error(f"Workspace {workspace_handle!r} not found.")
-
-                    update_current_profile(
-                        {
-                            PREFECT_API_URL: build_url_from_workspace(
-                                workspace_handle_details[workspace_handle]
-                            )
-                        }
-                    )
-                    current_workspace = workspace_handle
-
-            exit_with_success(
-                f"Logged in to Prefect Cloud using profile {profile_name!r}.\n"
-                f"Workspace is currently set to {current_workspace!r}. "
-                f"The workspace can be changed using `prefect cloud workspace set`."
+    if workspace_handle:
+        # Search for the given workspace
+        for workspace in workspaces:
+            if workspace.handle == workspace_handle:
+                break
+        else:
+            exit_with_error(f"Workspace {workspace_handle!r} not found.")
+    else:
+        # Prompt a switch if the number of workspaces is greater than one
+        prompt_switch_workspace = len(workspaces) > 1
+
+        current_workspace = get_current_workspace(workspaces)
+
+        # Confirm that we want to switch if the current profile is already logged in
+        if (
+            current_profile_is_logged_in or current_workspace is not None
+        ) and prompt_switch_workspace:
+            app.console.print(
+                f"You are currently using workspace {current_workspace.handle!r}."
+            )
+            prompt_switch_workspace = typer.confirm(
+                "? Would you like to switch workspaces?", default=False
             )
 
-    if not workspace_handle:
-        workspace_handle = select_workspace(workspace_handle_details.keys())
-
-    if workspace_handle not in workspace_handle_details:
-        exit_with_error(f"Workspace {workspace_handle!r} not found.")
+        if prompt_switch_workspace:
+            workspace = prompt_select_from_list(
+                app.console,
+                "Which workspace would you like to use?",
+                [(workspace, workspace.handle) for workspace in workspaces],
+            )
+        else:
+            workspace = current_workspace or workspaces[0]
 
-    current_profile = update_current_profile(
+    update_current_profile(
         {
-            PREFECT_API_URL: build_url_from_workspace(
-                workspace_handle_details[workspace_handle]
-            ),
             PREFECT_API_KEY: key,
-        },
+            PREFECT_API_URL: workspace.api_url(),
+        }
     )
 
     exit_with_success(
-        f"Logged in to Prefect Cloud using profile {current_profile.name!r}.\n"
-        f"Workspace is currently set to {workspace_handle!r}. "
-        f"The workspace can be changed using `prefect cloud workspace set`."
+        f"Authenticated with Prefect Cloud! Using workspace {workspace.handle!r}."
```

Comment:
```suggestion
        f"Logged in to the {workspace.handle!r} workspace in Prefect Cloud; use `prefect cloud workspace set` to log in to a different workspace."
```
Not 100% on this wording but it'd still be good to include a happy path for users that selected the wrong workspace.

Reply:
A few commits back, I conditionally displayed the `prefect cloud workspace set` hint depending on if they had more than one workspace. I can add that again.

Reply:
Ah I like that branching - makes sense to me